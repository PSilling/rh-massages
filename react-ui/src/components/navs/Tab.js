// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { NavItem, NavLink } from "reactstrap";

/**
 * Single labeled Tab component used in navigation tabbing.
 */
class Tab extends Component {
  shouldComponentUpdate(nextProps) {
    return this.props.active !== nextProps.active || this.props.label !== nextProps.label;
  }

  handleKeyPress = event => {
    if (event.key === "Enter") {
      this.props.onRemoveClick();
    }
  };

  render() {
    return (
      <NavItem>
        <NavLink
          style={this.props.active ? {} : { cursor: "pointer", color: "#888" }}
          className={this.props.active ? "active" : ""}
          onClick={this.props.active ? () => {} : this.props.onClick}
        >
          {this.props.label}
          {this.props.onRemoveClick !== null && this.props.active && (
            <span
              className="ml-2"
              style={{ cursor: "pointer" }}
              role="button"
              tabIndex="-2"
              onClick={this.props.onRemoveClick}
              onKeyPress={this.handleKeyPress}
            >
              &times;
            </span>
          )}
        </NavLink>
      </NavItem>
    );
  }
}

Tab.propTypes = {
  /** whether the tab is the active one */
  active: PropTypes.bool,
  /** tab title label */
  label: PropTypes.string,
  /** function called on tab click */
  onClick: PropTypes.func,
  /** function called on remove button click */
  onRemoveClick: PropTypes.func
};

Tab.defaultProps = {
  active: false,
  label: "Tab",
  onClick: null,
  onRemoveClick: null
};

export default Tab;
