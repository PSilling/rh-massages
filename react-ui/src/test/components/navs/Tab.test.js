// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import Tab from '../../../components/navs/Tab';

test('renders content with correct props', () => {
  const testFunction = jest.fn(),
        testRenderer = TestRenderer.create(
          <Tab label="test" onClick={testFunction} />
        ),
        testInstance = testRenderer.root;

  let list = testInstance.findByType('li'),
      link = testInstance.findByType('a'),
      treeJSON = testRenderer.toJSON();

  expect(testFunction).not.toHaveBeenCalled();

  link.props.onClick();

  expect(testFunction).toHaveBeenCalledTimes(1);
  expect(list.props.className).toBe("");
  expect(link.props.children).toEqual("test");
  expect(link.props.onClick).toBe(testFunction);
  expect(treeJSON).toMatchSnapshot();
});

test('activates Tab based on activation prop', () => {
  const testRenderer = TestRenderer.create(
          <Tab active label="active" />
        ),
        testInstance = testRenderer.root;

  let list = testInstance.findByType('li'),
      link = testInstance.findByType('a');

  expect(list.props.className).toBe("active");
  expect(link.props.children).toEqual("active");
});
