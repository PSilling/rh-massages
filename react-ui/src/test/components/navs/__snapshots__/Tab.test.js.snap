// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
<li
  className=""
>
  <a
    onClick={[Function]}
    style={
      Object {
        "color": "#888",
        "cursor": "pointer",
        "fontSize": "16px",
      }
    }
  >
    test
  </a>
</li>
`;
